components/studio/libexec/hab-studio-type-default.sh

```diff
@@ -88,17 +85,7 @@ finish_setup() {
   # (This is also why we're not using HAB_BLDR_CHANNEL for this and
   # replicating the fallback logic from hab-plan-build; it'd be too
   # easy to create an unstable studio.)
-  for pkg in $pkgs; do
-    if [ -n "${CI_OVERRIDE_CHANNEL:-}" ]; then
-      info "Override channel set; retrieving ${pkg} from ${CI_OVERRIDE_CHANNEL}"
-      _hab install --channel="${CI_OVERRIDE_CHANNEL}" "${pkg}" || {
-        info "Package not found in ${CI_OVERRIDE_CHANNEL}; falling back to stable"
-        _hab install "${pkg}"
-      }
-    else
-      _hab install "$pkg"
-    fi
-  done
+  _hab install $HAB_STUDIO_BACKLINE_PKG
```

Comment:
Can you explain a little why `CI_OVERRIDE_CHANNEL` is no longer necessary? I'm happy to see it go, I just don't understand why.

Reply:
Before this change, core/hab-backline was an implicit dependency of the studio used when the studio is invoked.  Since it's not declared, the studio would have to determine what backline package to use every time it is run.  This would always use the latest stable version of the package that matches `core/hab-backline/VERSION`. In the context of a release, we don't want the latest stable but we can't rely on HAB_BLDR_CHANNEL as that has a different intent and overloading it to test backline caused issues (Note: I wasn't around at the time so I don't know the specifics) so `CI_OVERRIDE_CHANNEL` was introduced.

By adding `core/hab-backline` to `pkg_deps`, we can leverage the plan-build logic to always resolve that to a fully qualified ident at build time, which allows us to set `$HAB_STUDIO_BACKLINE_PKG` at the time of build. With the change to `studio enter` to set up the environment as specified by the studio package, we know exactly what backline package the studio was built with and can correctly install it on the _inside_ of the studio. 

This is in alignment with how packages in Habitat are intended to work. Rather than the existing behavior where we've built a back door in order to test changes to a component, we now rebuild the dependent packages against the new version and `CI_OVERRIDE_CHANNEL` can go away.

Reply:
This is my favorite thing!